pkg/hub/client/policy_merger_test.go

```diff
@@ -331,6 +331,90 @@ func TestPolicyEnforcedClient_mergePolicies(t *testing.T) {
 			},
 			order: policy.MergeOrder{policy.Action, policy.Global},
 		},
+		{
+			name: "merge type instances and additional input",
+			action: policy.ActionPolicy{
+				Rules: policy.RulesList{
+					policy.RulesForInterface{
+						Interface: types.ManifestRef{
+							Path: interfacePath,
+						},
+						OneOf: []policy.Rule{
+							{
+								ImplementationConstraints: policy.ImplementationConstraints{
+									Path: &implementationPath,
+								},
+								Inject: &policy.InjectData{
+									AdditionalInput: map[string]interface{}{
+										"additional-input": map[string]interface{}{
+											"a": 1,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			global: policy.Policy{
+				Rules: policy.RulesList{
+					policy.RulesForInterface{
+						Interface: types.ManifestRef{
+							Path: interfacePath,
+						},
+						OneOf: []policy.Rule{
+							{
+								ImplementationConstraints: policy.ImplementationConstraints{
+									Path: &implementationPath,
+								},
+								Inject: &policy.InjectData{
+									TypeInstances: []policy.TypeInstanceToInject{
+										{
+											ID: "123-321-123",
+											TypeRef: types.ManifestRef{
+												Path: "cap.type.x",
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expected: policy.Policy{
+				Rules: policy.RulesList{
+					policy.RulesForInterface{
+						Interface: types.ManifestRef{
+							Path: interfacePath,
+						},
+						OneOf: []policy.Rule{
+							{
+								ImplementationConstraints: policy.ImplementationConstraints{
+									Path: &implementationPath,
+								},
+								Inject: &policy.InjectData{
+									TypeInstances: []policy.TypeInstanceToInject{
+										{
+											ID: "123-321-123",
+											TypeRef: types.ManifestRef{
+												Path: "cap.type.x",
+											},
+										},
+									},
+									AdditionalInput: map[string]interface{}{
+										"additional-input": map[string]interface{}{
+											"a": 1,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			order: policy.MergeOrder{policy.Action, policy.Global},
+		},
```

Comment:
Do we want to also test merging nested additional Input (global + action workflow)? It's quite similar to the nestedWorkflowPolicy, but it is not equal